### PR TITLE
Fix santizer test failure with dynamic linking

### PR DESCRIPTION
**Pull request by sbc100**
I thought I had fixed all these #17659 but I was clearly not running
all of the tests.

### CODE REVIEW

File: test/dlmalloc_proxy.c [hunk 1]
```diff
@@ -43,9 +43,10 @@ void unget_lib() {
 int main() {
   int n = 0, total = 0, l = 0;
   void *allocs[50];
-  allocs[10] = malloc(10); // pull in real malloc
+  allocs[n++] = malloc(10); // pull in real malloc
```
Comment (sbc100) [inline test/dlmalloc_proxy.c:46]: This is a fix for a genuine leak

File: emscripten.py [hunk 2]
```diff
@@ -679,7 +679,7 @@ def make_export_wrappers(exports, delay_assignment):
     # The emscripten stack functions are called very early (by writeStackCookie) before
     # the runtime is initialized so we can't create these wrappers that check for
     # runtimeInitialized.
-    if settings.ASSERTIONS and not name.startswith('emscripten_stack_'):
+    if settings.ASSERTIONS and not name.startswith('emscripten_stack_') and name != '__trap':
```
Comment (kripken) [inline emscripten.py:682]: What is `__trap`? A comment here would be good to add to the existing comment about stack functions.

**Review by kripken — COMMENTED** [review]
lgtm % that question

File: emscripten.py (hunk 2 above)
Comment (sbc100) [reply emscripten.py:684]: Added comment.

**Review by kripken — APPROVED** [review]
Makes sense, thanks!

**Pull request merged**